### PR TITLE
feat(): add feedbacks controller, create method and route

### PR DESCRIPTION
**Pull request by jgmontoya**
Also changes the response enum from positive and negative to gift_found and gift_not_found to avoid confusion.

### CODE REVIEW

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            end
+
+            it 'updates the givers email' do
+                post :create, params: gift_not_found_feedback_params, as: :json
+                expect(giver.reload.email).to eq(email)
+            end
+        end
+
+    end
+
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Extra empty line detected at block body end.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                post :create, params: gift_not_found_feedback_params, as: :json
+                expect(giver.reload.phone_number).to eq(phone_number)
+            end
+
+            it 'updates the givers email' do
+                post :create, params: gift_not_found_feedback_params, as: :json
+                expect(giver.reload.email).to eq(email)
+            end
+        end
+
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Extra empty line detected at block body end.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                expect(response).to have_http_status(:success)
+            end
+
+            it 'updates the givers phone number' do
+                post :create, params: gift_not_found_feedback_params, as: :json
+                expect(giver.reload.phone_number).to eq(phone_number)
+            end
+
+            it 'updates the givers email' do
+                post :create, params: gift_not_found_feedback_params, as: :json
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 (not 4) spaces for indentation.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        end
+
+        context 'when response is gift_not_found' do
+            it 'returns http success' do
+                post :create, params: gift_not_found_feedback_params, as: :json
+                expect(response).to have_http_status(:success)
+            end
+
+            it 'updates the givers phone number' do
+                post :create, params: gift_not_found_feedback_params, as: :json
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 (not 4) spaces for indentation.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        context 'when response is gift_found' do
+            it 'returns http success' do
+                post :create, params: gift_found_feedback_params, as: :json
+                expect(response).to have_http_status(:success)
+            end
+        end
+
+        context 'when response is gift_not_found' do
+            it 'returns http success' do
+                post :create, params: gift_not_found_feedback_params, as: :json
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 (not 4) spaces for indentation.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+        context 'when response is gift_found' do
+            it 'returns http success' do
+                post :create, params: gift_found_feedback_params, as: :json
+                expect(response).to have_http_status(:success)
+            end
+        end
+
+        context 'when response is gift_not_found' do
+            it 'returns http success' do
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 (not 4) spaces for indentation.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            } }
+        end
+
+        before do
+            session['receiver_id'] = receiver.id
+        end
+
+        context 'when response is gift_found' do
+            it 'returns http success' do
+                post :create, params: gift_found_feedback_params, as: :json
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 (not 4) spaces for indentation.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                    email: email
+            } }
+        end
+
+        before do
+            session['receiver_id'] = receiver.id
+        end
+
+        context 'when response is gift_found' do
+            it 'returns http success' do
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 (not 4) spaces for indentation.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+            { feedback: {
+                    response: 'gift_not_found',
+                    additional_info: 'This is additional information regarding my gift.',
+                    phone_number: phone_number,
+                    email: email
+            } }
+        end
+
+        before do
+            session['receiver_id'] = receiver.id
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 (not 4) spaces for indentation.

File: spec/controllers/api/v1/feedbacks_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        let(:gift_found_feedback_params) do
+            { feedback: {
+                response: 'gift_found'
+            } }
+        end
+        let(:phone_number) { '+5691234567' }
+        let(:email) { 'example_email@example.com' }
+        let(:gift_not_found_feedback_params) do
+            { feedback: {
+                    response: 'gift_not_found',
```
Comment (monkeyci) [inline spec/controllers/api/v1/feedbacks_controller_spec.rb]: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    end
+
+    def feedback_params
+      params.require(:feedback).permit(:response, :additional_info)
+    end
+
+    def contact_params
+      params.require(:feedback).permit(:email, :phone_number)
+    end
+  end
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: end at 29, 2 is not aligned with class at 1, 0.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        response: feedback_params[:response],
+        additional_info: feedback_params[:additional_info]
+      )
+    end
+
+    def feedback_params
+      params.require(:feedback).permit(:response, :additional_info)
+    end
+
+    def contact_params
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: Use 2 (not 4) spaces for indentation.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+    def save_feedback
+      Feedback.create!(
+        receiver_id: receiver.id,
+        response: feedback_params[:response],
+        additional_info: feedback_params[:additional_info]
+      )
+    end
+
+    def feedback_params
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: Use 2 (not 4) spaces for indentation.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+      receiver.giver.update!(contact_params)
+    end
+
+    def save_feedback
+      Feedback.create!(
+        receiver_id: receiver.id,
+        response: feedback_params[:response],
+        additional_info: feedback_params[:additional_info]
+      )
+    end
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: end at 20, 6 is not aligned with def at 14, 4.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+
+    private
+
+    def update_contact_info
+      receiver.giver.update!(contact_params)
+    end
+
+    def save_feedback
+      Feedback.create!(
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: Use 2 (not 4) spaces for indentation.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+      respond_with feedback
+    end
+
+    private
+
+    def update_contact_info
+      receiver.giver.update!(contact_params)
+    end
+
+    def save_feedback
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: Use 2 (not 4) spaces for indentation.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 17]
```diff
@@ -0,0 +1,29 @@
+class Api::V1::FeedbacksController < Api::V1::BaseController
+    def create
+      feedback = save_feedback
+      update_contact_info if feedback_params[:response] == 'gift_not_found'
+      respond_with feedback
+    end
+
+    private
+
+    def update_contact_info
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: Use 2 (not 4) spaces for indentation.

File: app/controllers/api/v1/feedbacks_controller.rb [hunk 18]
```diff
@@ -0,0 +1,29 @@
+class Api::V1::FeedbacksController < Api::V1::BaseController
+    def create
```
Comment (monkeyci) [inline app/controllers/api/v1/feedbacks_controller.rb]: Use 2 (not 4) spaces for indentation.

File: app/serializers/feedback_serializer.rb [hunk 19]
```diff
@@ -0,0 +1,5 @@
+class FeedbackSerializer < ActiveModel::Serializer
+    include Rails.application.routes.url_helpers
+
+    attributes :id, :receiver_id, :response, :additional_info
+  end
```
Comment (monkeyci) [inline app/serializers/feedback_serializer.rb]: end at 5, 2 is not aligned with class at 1, 0.

File: app/serializers/feedback_serializer.rb [hunk 20]
```diff
@@ -0,0 +1,5 @@
+class FeedbackSerializer < ActiveModel::Serializer
+    include Rails.application.routes.url_helpers
+
+    attributes :id, :receiver_id, :response, :additional_info
```
Comment (monkeyci) [inline app/serializers/feedback_serializer.rb]: Use 2 (not 4) spaces for indentation.

File: app/serializers/feedback_serializer.rb [hunk 21]
```diff
@@ -0,0 +1,5 @@
+class FeedbackSerializer < ActiveModel::Serializer
+    include Rails.application.routes.url_helpers
```
Comment (monkeyci) [inline app/serializers/feedback_serializer.rb]: Use 2 (not 4) spaces for indentation.

File: app/serializers/feedback_serializer.rb [hunk 22]
```diff
@@ -0,0 +1,5 @@
+class FeedbackSerializer < ActiveModel::Serializer
+  include Rails.application.routes.url_helpers
```
Comment (imarquezc) [inline app/serializers/feedback_serializer.rb]: qué hace esta línea? es necesaria?

Comment (jgmontoya) [reply app/serializers/feedback_serializer.rb]: Se me pasó, no es necesaria! la elimino al tiro

File: app/serializers/feedback_serializer.rb [hunk 23]
```diff
@@ -0,0 +1,4 @@
+class FeedbackSerializer < ActiveModel::Serializer
+
```
Comment (monkeyci) [inline app/serializers/feedback_serializer.rb]: Extra empty line detected at class body beginning.

**Pull request merged**